README.md

```diff
@@ -21,94 +21,175 @@ Pipeline.create(...)
 ## Usage
 
 #### Pipeline
- 
-A `registerScalaCoders` methods has been added to `Pipeline`. This adds Coders for the following types:
-  
-  * `Int`, `Long`, `Double`
-  * `Option`, `Try`, `Either`
-  * `Tuple2` to `Tuple22`
 
-and then returns the pipeline itself so that it can be invoked fluently.
+`Pipeline` has been enriched with a handful of methods.
 
-In addition a `run` method has been added to the `POutput` type, so that if you have a pipeline with a single final stage you can invoke it directly. 
+To create a `PCollection` from in-memory data use the `transform` method instead of `apply`. This method ensures that the coder is set correctly on the input data. 
+ 
+In addition a `run` method has been added to the `POutput` type, so that you can fluently chain transforms then run your pipeline. For example:
     
 ```scala
-   val result = Pipeline.create(...)
-     .registerScalaCoders()
-     .apply(... /* transforms*/)
-     .apply(... /* more transforms*/)
-     .run()
+val result = Pipeline.create(...)
+  .transform(Create.of("foo", "bar"))
+  .apply(...transforms...)
+  .run()
 ```
   
-#### Basic Collections Methods
+#### Basic PCollection Methods
   
-`map`, `flatMap`, `filter` and `collect` methods have been added to `PCollection`. Each of these methods behaves identically to the standard Scala collections, with the notable exception of `flatMap`. This method turns each elements of the input `PCollection` into an `Iterable` (possibly of a different type) that is then flattened into the output `PCollection`.
+`PCollection` now has `map`, `flatMap`, `filter` and `collect` methods that each behave as you would expect.
     
-Simple example that puts it all together:
+Simple example:
+
+```scala
+val result = Pipeline.create(...)
+  .transform(Create.of("123", "456", "789"))
+  .flatMap(_.split(""))
+  .map(_.toInt)
+  .filter(_ < 5)
+  .collect {
+    case x if x % 3 == 0 => if (x % 5 == 0) "FizzBuzz" else "Fizz"
+    case x if x % 5 == 0 => "Buzz"
+  }
+  .run()
+```
+
+#### PCollection Extras
+
+##### Logging Side Effect
+
+A side-effecting method `foreach` has been added in order to allow handy debug logging. This method supplies each element of the `PCollection` to it's argument then passes on the element unchanged.
+For example:
+
+```scala
+val result = Pipeline(...)
+  .transform(Create.of("123", "456", "789"))
+  .foreach(println)
+  .apply(...continue as normal...) 
+```
+
+##### Extracting Timetamps
+
+`extractTimestamp` converts each element in the PCollection to a tuple with its corresponding timestamp. For example:
+```scala
+val collection: PCollection[(String, Instant)] = Pipeline.create(...)
+  .transform(Create.of("foo", "bar"))
+  .withTimestamps
+```
+
+##### Converting to a `KV`
+
+The `withKey` method provides a drop in replacement for the `WithKeys` transform.
+
+##### Joining PCollections of the same type
```

Comment:
I think we should avoid using the term "join" here, as that implies that you'll get a relational join – how about "merge" instead?